packages/mdc-top-app-bar/foundation.ts

```diff
@@ -50,64 +50,23 @@ export class MDCTopAppBarBaseFoundation extends MDCFoundation<MDCTopAppBarAdapte
       hasClass: () => false,
       setStyle: () => undefined,
       getTopAppBarHeight: () => 0,
-      registerNavigationIconInteractionHandler: () => undefined,
-      deregisterNavigationIconInteractionHandler: () => undefined,
       notifyNavigationIconClicked: () => undefined,
-      registerScrollHandler: () => undefined,
-      deregisterScrollHandler: () => undefined,
-      registerResizeHandler: () => undefined,
-      deregisterResizeHandler: () => undefined,
       getViewportScrollY: () => 0,
       getTotalActionItems: () => 0,
     };
     // tslint:enable:object-literal-sort-keys
   }
 
-  protected scrollHandler_?: SpecificEventListener<'scroll'>;
-  protected resizeHandler_?: SpecificEventListener<'resize'>;
-  private readonly navClickHandler_: SpecificEventListener<'click'>;
+  handleTargetScroll?: SpecificEventListener<'scroll'>;
```

Comment:
Again couldn't comment on previous thread.

`handleTargetScroll` needs to be a abstract method which can be overriden in child classes.

Reply:
Please see `mdc-drawer/dismissible/foundation.ts` for example where `opened()` & `closed()` are abstract methods in Dismissible variant where Modal foundation overrides it.

Reply:
I do not see any abstract methods in the dismissible/foundation. I see `protected`, but the top-app-bar/component needs to call the `handleX` methods.

Also if they are abstract, that means they technically don't always exist, which means I'll need the check I am removing RE [this comment](https://github.com/material-components/material-components-web/pull/4701#discussion_r284412039), correct?

Reply:
abstract method is just an empty function (which is a truthy value), in typescript we use `abstract` method to enforce child classes to implement.

Reply:
I'm getting error `Abstract methods can only appear within an abstract class.`. I think I just make them empty methods.

Reply:
spoke offline...will not have abstract methods, but instead will just have noops